### PR TITLE
MCO: runs hypershift on openstack jobs as optional

### PR DESCRIPTION
**Pull request by EmilienM**
That will help us to test this scenario when sending PRs in MCO.

### CODE REVIEW

**Comment by EmilienM** [comment]
/pj-rehearse pull-ci-openshift-machine-config-operator-master-e2e-openstack-hypershift

**Comment by mdbooth** [comment]
I'm very much in favour of this once we've tested it passes... which it won't because of https://github.com/openshift/machine-config-operator/pull/4611. Bah, chicken and egg. Lets at least see it spawn a cluster and fail in the expected manner.

**Comment by EmilienM** [comment]
@mdbooth my rehearse it to make sure I didn't miss e.g. an image in the list of images needed in the MCO jobs config to run HCP on OSP. But yes the job will fail and it's fine.

**Comment by EmilienM** [comment]
> @mdbooth my rehearse it to make sure I didn't miss e.g. an image in the list of images needed in the MCO jobs config to run HCP on OSP. But yes the job will fail and it's fine.

like this one:

```
 ERRO[2024-09-24T19:50:53Z] Failed to load arguments.                     error=tests[e2e-openstack-hypershift].steps.test[3].from: unknown image "test-bin" (configuration is missing `test_binary_build_commands`) 
```

**Comment by EmilienM** [comment]
/pj-rehearse pull-ci-openshift-machine-config-operator-master-e2e-openstack-hypershift

**Comment by EmilienM** [comment]
/pj-rehearse pull-ci-openshift-machine-config-operator-master-e2e-openstack-hypershift

**Comment by EmilienM** [comment]
The error in the job is expected and it has reached e2e so we're good.

/pj-rehearse ack

File: ci-operator/step-registry/hypershift/openstack/e2e/execute/hypershift-openstack-e2e-execute-ref.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ ref:
       name: release:initial
     - env: CI_HYPERSHIFT_OPERATOR
       name: pipeline:hypershift-operator
-  from: test-bin
+  from: hypershift-tests
```
Comment (MaysaMacedo) [inline ci-operator/step-registry/hypershift/openstack/e2e/execute/hypershift-openstack-e2e-execute-ref.yaml:16]: why this change?

**Review by MaysaMacedo — COMMENTED** [review]
/lgtm

I just had one question

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 2]
```diff
@@ -2756,6 +2756,85 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-openstack-externallb,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (yuqi-zhang) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:2760]: Could I ask the intent behind this job? Most PRs should be covered by the existing hypershift job + regular MCO jobs, so I don't think we would need to always_run this

**Review by yuqi-zhang — COMMENTED** [review]
Mostly lgtm, one question inline

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml (hunk 2 above)
Comment (EmilienM) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:2760]: that's a mistake. Sorry, let me fix this

File: ci-operator/step-registry/hypershift/openstack/e2e/execute/hypershift-openstack-e2e-execute-ref.yaml (hunk 1 above)
Comment (EmilienM) [reply ci-operator/step-registry/hypershift/openstack/e2e/execute/hypershift-openstack-e2e-execute-ref.yaml:16]: I have to, the test-bin image is only in Hypershift jobs config. hypershift-tests image is the one that we should use I think.
